src/algo/single_source/single_source.py

```diff
@@ -15,7 +16,8 @@ def __init__(self, graph: LabelGraph, grammar: CnfGrammar):
 
     def init_simple_rules(self):
         for l, r in self.grammar.simple_rules:
-            self.sources[l] = self.graph[r].dup()
+            self.nonterms[l] = self.graph[r].dup()
```

Comment:
Check what happens if there are 2 simple rules with the same left part (A -> a, A -> b).